### PR TITLE
chore/add JsDoc for address.ts file

### PR DESCRIPTION
**Pull request by BlackStarkGoku**
I simply added JSDocs for address.ts file, won't affect tests or code. If everything is alright, I can add more.

### CODE REVIEW

**Review by ivpavici — COMMENTED** [review]
looks like a good direction!
please resolve merge conflicts :)

File: src/utils/address.ts [hunk 1]
```diff
@@ -11,6 +11,12 @@ import { assertInRange, toHex } from './num';
  * Format a hex number to '0x' and 64 characters, adding leading zeros if necessary.
  * @param {BigNumberish} address
  * @returns {string} Hex string : 0x followed by 64 characters. No upper case characters in the response.
+ * @example
+ * ```typescript
+ * const address = 0x7ee790591d9fa3efc87067d95a643f8455e0b8190eb8cb7bfd39e4fb7571fdf;
+ * const result = addAddressPadding(address);
+ * // result = "0x07ee790591d9fa3efc87067d95a643f8455e0b8190eb8cb7bfd39e4fb7571fdf"
```
Comment (PhilippeR26) [inline src/utils/address.ts:18]: Address is a string. To use between quotes.
You should use an address that needs more zeros (for example 3), as we have a function that optionally add a zero to have an odd number of characters.

**Comment by PhilippeR26** [comment]
Target branch has to be `next-version`

File: src/utils/address.ts [hunk 2]
```diff
@@ -65,6 +82,12 @@ export function getChecksumAddress(address: BigNumberish): string {
  * @param address string
  *
  * @returns true if the ChecksumAddress is valid
+ * @example
+ * ```typescript
+ * const address = "0x0000090591D9fA3EfC87067d95a643f8455E0b8190eb8Cb7bFd39e4fb7571fDF";
+ * const result = validateChecksumAddress(address);
+ * // result = "true"
```
Comment (PhilippeR26) [inline src/utils/address.ts:89]: Result is a boolean, not a string.
So, double quotes have to be removed.

File: src/utils/encode.ts
Comment (PhilippeR26) [inline src/utils/encode.ts:1]: Encode functions are accessible for the users by using the namespace `encode`.
The examples have to be usable by the users.
So, you have to add in all examples in this file  : `encode.` before each use of an encoding function .

Comment (PhilippeR26) [reply src/utils/encode.ts:1]: To be consistent in all the project, use `* // result = ` to display the result (as you made perfectly in `address.ts`).

File: src/utils/encode.ts [hunk 3]
```diff
@@ -14,6 +14,15 @@ const STRING_ZERO = '0';
  * Convert array buffer to string
  *
  * *[internal usage]*
+ *
+ * @param {ArrayBuffer} array The ArrayBuffer to convert to string.
+ * @returns {string} The converted string.
+ *
+ * @example
+ * ```typescript
+ * const string = arrayBufferToString({0: 72, 1: 105});
```
Comment (PhilippeR26) [inline src/utils/encode.ts:23]: This example isn't working :
```
error TS2353: Object literal may only specify known properties, and '0' does not exist in type 'ArrayBuffer'.
const string = encode.arrayBufferToString({0: 72, 1: 105});
                                                                     Î
```

File: src/utils/encode.ts [hunk 4]
```diff
@@ -23,6 +32,16 @@ export function arrayBufferToString(array: ArrayBuffer): string {
...
  * *[internal usage]*
+ *
+ * @param {string} str The UTF-8 string to convert.
+ * @returns {Uint8Array} The encoded Uint8Array.
+ *
+ * @example
+ * ```typescript
+ * const myString = 'Hi';
+ * const byteArray = utf8ToArray(myString);
+ * console.log(byteArray); // Outputs: Uint8Array {0: 72, 1: 105}
```
Comment (PhilippeR26) [inline src/utils/encode.ts:43]: Have you tested your example?
The result is not a struct, but an array (as the name is explaining) :
```
[ 72, 105 ]
```

File: src/utils/encode.ts [hunk 5]
```diff
@@ -39,13 +58,33 @@ export function stringToArrayBuffer(str: string): Uint8Array {
...
  * Convert string to array buffer (browser and node compatible)
+ *
+ * @param {string} a The Base64 encoded string to convert.
+ * @returns {Uint8Array} The decoded Uint8Array.
+ *
+ * @example
+ * ```typescript
+ * const base64String = 'SGVsbG8='; // 'Hello' in Base64
+ * const byteArray = atobUniversal(base64String);
+ * console.log(byteArray); // Outputs: Uint8Array {0: 72, 1: 105, 2: 108, 3: 108, 4: 111}
```
Comment (PhilippeR26) [inline src/utils/encode.ts:69]: Response is not an object, but an array : 
```
[ 72, 101, 108, 108, 111 ]
```

File: src/utils/encode.ts [hunk 6]
```diff
@@ -39,13 +58,33 @@ export function stringToArrayBuffer(str: string): Uint8Array {
...
+ * ```
  */
 export function atobUniversal(a: string): Uint8Array {
   return base64.decode(a);
 }
 
 /**
  * Convert array buffer to string (browser and node compatible)
+ *
+ * @param {Uint8Array} b The decoded Uint8Array.
```
Comment (PhilippeR26) [inline src/utils/encode.ts:79]: ArrayBuffer

File: src/utils/encode.ts [hunk 7]
```diff
@@ -54,6 +93,13 @@ export function btoaUniversal(b: ArrayBuffer): string {
 /**
  * Convert array buffer to hex-string
  * @returns format: hex-string
```
Comment (PhilippeR26) [inline src/utils/encode.ts:95]: param missing.

File: src/utils/encode.ts [hunk 8]
```diff
@@ -54,6 +93,13 @@ export function btoaUniversal(b: ArrayBuffer): string {
 /**
  * Convert array buffer to hex-string
```
Comment (PhilippeR26) [inline src/utils/encode.ts:94]: param missing.

File: src/utils/encode.ts (hunk 7 above)
Comment (PhilippeR26) [inline src/utils/encode.ts:95]: Could you add that it's without "0x"

File: src/utils/encode.ts [hunk 9]
```diff
@@ -81,6 +141,29 @@ export function addHexPrefix(hex: string): string {
...
+ *
+ * const paddedRight = padString(myString, 10, false);
+ * console.log(paddedRight); // Outputs: 'hello00000'
+ *
+ * // Using a different padding character
+ * const paddedCustom = padString(myString, 10, true, '1');
+ * console.log(paddedCustom); // Outputs: '11111hello'
+ * ```
  */
 function padString(str: string, length: number, left: boolean, padding = STRING_ZERO): string {
```
Comment (PhilippeR26) [inline src/utils/encode.ts:168]: This function is not exported, so JSDoc isn't really useful.
But it's made, so keep it (without `encode.` prefix, as it's not exported).

File: src/utils/encode.ts [hunk 10]
```diff
@@ -94,6 +177,21 @@ function padString(str: string, length: number, left: boolean, padding = STRING_
...
+ * The padding can be applied only to the left of the input string.
+ *
+ * @param {string} str The string to pad.
+ * @param {number} length The target length for the padded string.
+ * @param {string} [padding='0'] The string to use for padding. Defaults to '0'.
+ * @returns {string} The padded string.
+ *
+ * @example
+ * ```typescript
+ * const myString = 'hello';
```
Comment (PhilippeR26) [inline src/utils/encode.ts:191]: This function is used for hex string, not for text. Please change to a hex num example.

File: src/utils/encode.ts
Comment (PhilippeR26) [reply src/utils/encode.ts:1]: Each parameter that has a default value has to be written with :
```
@param {type} [name='0']
```

File: src/utils/encode.ts [hunk 11]
```diff
@@ -125,6 +259,13 @@ export function sanitizeBytes(str: string, byteSize = 8, padding = STRING_ZERO):
  * *[no internal usage]*
  * @param hex hex-string
  * @returns format: hex-string
+ *
+ * @example
+ * ```typescript
+ * const unevenHex = '0x23abc';
+ * const formattedHex = sanitizeHex(unevenHex);
+ * console.log(formattedHex); // Outputs: '0x023abc' (padded to ensure even length)
```
Comment (PhilippeR26) [inline src/utils/encode.ts:267]: Could you modify the definition above (line 257) to explain the even length?

**Review by PhilippeR26 — CHANGES_REQUESTED** [review]
Several things to correct (some comments are valid for the whole encode.ts file).

File: src/utils/encode.ts (hunk 4 above)
Comment (PhilippeR26) [reply src/utils/encode.ts:43]: maybe, but ts-node is reporting :
image
So, Uint8Array is an array... sounds so logical...

File: src/utils/encode.ts (hunk 5 above)
Comment (PhilippeR26) [reply src/utils/encode.ts:69]: image

**Review by PhilippeR26 — CHANGES_REQUESTED** [review]
One change requested.

**Review by PhilippeR26 — APPROVED** [review]
@ivpavici Finally lgfm

**Review by ivpavici — APPROVED** [review]

**Pull request merged**